form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowDeploymentService.kt

```diff
@@ -64,14 +64,18 @@ class FormFlowDeploymentService(
     fun deploy(formFlowKey: String, formFlowJson: String) {
         validate(formFlowJson)
 
-        val formFlowDefinition = Mapper.get().readValue(formFlowJson, FormFlowDefinition::class.java)
+        val formFlowDefinitionConfig = Mapper.get()
+            .readValue(
+                formFlowJson,
+                com.ritense.formflow.domain.definition.configuration.FormFlowDefinition::class.java
+            )
 
         try {
             val existingDefinition = formFlowService.findLatestDefinitionByKey(formFlowKey)
             var definitionId = FormFlowDefinitionId.newId(formFlowKey)
 
             if (existingDefinition != null) {
-                if (formFlowDefinition == existingDefinition) {
+                if (formFlowDefinitionConfig.contentEquals(existingDefinition)) {
```

Comment:
There's a difference now between the loaded configuration and the stored definitions. We split them up because we were having to write bad code to make the deserialization and use as entity working at the same time.

Now because these are different classes we can't use the default equals method, because that would go against the java best practices.